### PR TITLE
Add option to disable creeper exploding blocks

### LINKED ISSUE #199: Creeper explode/don't explode blocks
### Info

This is:
- [ ] bug report
- [X] feature request
- [ ] other

### Your issue / suggestion
Add an option in **config.yml** for the **creeper to explode blocks or not to explode**, preventing the creeper grief!

### Configuration
Config:
👇 Not to explode blocks 👇
```
creeper-explode-blocks: false
```
👇 For explode blocks 👇
```
creeper-explode-blocks: true
```

### PR DESCRIPTION
**Pull request by PleaseInsertNameHere**
This pull request adds an option to disable creeper exploding blocks - closes #199

### CODE REVIEW

File: src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java [hunk 1]
```diff
@@ -85,7 +86,7 @@ public void explode() {
         if (!ev.isCancelled()) {
             Explosion explosion = new Explosion(this, (float) ev.getForce(), this);
 
-            if (ev.isBlockBreaking() && this.level.getGameRules().getBoolean(GameRule.MOB_GRIEFING)) {
+            if (ev.isBlockBreaking() && this.level.getGameRules().getBoolean(GameRule.MOB_GRIEFING) && MobPlugin.getInstance().config.pluginConfig.getBoolean("creeper-explode-blocks")) {
```
Comment (PetteriM1) [inline src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java:89]: Please do not use pluginConfig here

File: src/main/java/nukkitcoders/mobplugin/Config.java [hunk 2]
```diff
@@ -20,10 +20,12 @@
     }
 
     boolean init(MobPlugin plugin) {
-        int ver = 15;
+        int ver = 16;
 
         if (pluginConfig.getInt("config-version") != ver) {
-            if (pluginConfig.getInt("config-version") == 14) {
+            if (pluginConfig.getInt("config-version") == 15) {
+                pluginConfig.set("creeper-explode-blocks", true);
```
Comment (PetteriM1) [inline src/main/java/nukkitcoders/mobplugin/Config.java:27]: Due to how the config updating is currently handled you should add this to every old config version we can update

File: src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java (hunk 1 above)
Comment (PetteriM1) [inline src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java:89]: I think using ev.setBlockBreaking(false) would be a better solution

**Review by PetteriM1 — CHANGES_REQUESTED** [review]

File: src/main/java/nukkitcoders/mobplugin/Config.java (hunk 2 above)
Comment (PleaseInsertNameHere) [reply src/main/java/nukkitcoders/mobplugin/Config.java:28]: Sorry, what do I have to change here?

Comment (PetteriM1) [reply src/main/java/nukkitcoders/mobplugin/Config.java]: `pluginConfig.set("creeper-explode-blocks", true);` for all old config versions, not just 15

File: src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java (hunk 1 above)
Comment (PetteriM1) [reply src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java]: And do not use pluginConfig here

File: src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java [hunk 3]
```diff
@@ -85,6 +86,10 @@ public void explode() {
         if (!ev.isCancelled()) {
             Explosion explosion = new Explosion(this, (float) ev.getForce(), this);
 
+            if (MobPlugin.getInstance().config.creeperExplodeBlocks) {
+                ev.setBlockBreaking(false);
```
Comment (PetteriM1) [inline src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java:90]: This should be done before the event is called on line 84

**Review by PetteriM1 — CHANGES_REQUESTED** [review]

File: src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java (hunk 3 above)
Comment (PleaseInsertNameHere) [reply src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java:90]: Oops, didn't see. Give me a second.

Comment (PetteriM1) [reply src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java]: You still do it after `callEvent`

Comment (PleaseInsertNameHere) [reply src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java]: Oops, sorry. Fixed it.

File: src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java [hunk 4]
```diff
@@ -80,6 +81,11 @@ public void explode() {
         if (this.closed) return;
 
         EntityExplosionPrimeEvent ev = new EntityExplosionPrimeEvent(this, this.isPowered() ? 6 : 3);
+
+        if (MobPlugin.getInstance().config.creeperExplodeBlocks) {
```
Comment (PetteriM1) [inline src/main/java/nukkitcoders/mobplugin/entities/monster/walking/Creeper.java:85]: Either you forgot the `!` or the boolean should be called creepers DON'T explode blocks

File: src/main/resources/config.yml [hunk 5]
```diff
@@ -75,3 +75,5 @@ other:
   end-enderman-spawning: 10
   # if tamed entity owner attacked or attack entity, it will be angry
   check-tamed-entity-attack: true
+  # disable creeper explode blocks
```
Comment (PetteriM1) [inline src/main/resources/config.yml:78]: Disable?

**Comment by PleaseInsertNameHere** [comment]
Alright, changed it. Should be better now.

**Pull request merged**